File: scripts/firebase_paths.py
"""Paths canônicos Firebase (Firestore + Storage) para uso em scripts Python.

Regra: nunca hardcode de tenant; sempre passar church_id dinamicamente.
"""


class FirebasePaths:
    # --- Firestore geral ---
    @staticmethod
    def igreja(church_id: str) -> str:
        return f"igrejas/{church_id.strip()}"

    @staticmethod
    def departamentos(church_id: str) -> str:
        return f"{FirebasePaths.igreja(church_id)}/departamentos"

    @staticmethod
    def departamento_doc(church_id: str, dept_id: str) -> str:
        return f"{FirebasePaths.departamentos(church_id)}/{dept_id.strip()}"

    @staticmethod
    def membros(church_id: str) -> str:
        return f"{FirebasePaths.igreja(church_id)}/membros"

    @staticmethod
    def config_mercado_pago(church_id: str) -> str:
        return f"{FirebasePaths.igreja(church_id)}/config/mercado_pago"

    # --- Financeiro ---
    @staticmethod
    def finance(church_id: str) -> str:
        return f"{FirebasePaths.igreja(church_id)}/finance"

    @staticmethod
    def finance_doc(church_id: str, finance_id: str) -> str:
        return f"{FirebasePaths.finance(church_id)}/{finance_id.strip()}"

    @staticmethod
    def finance_logs(church_id: str) -> str:
        return f"{FirebasePaths.igreja(church_id)}/finance_logs"

    @staticmethod
    def finance_mp_notifications(church_id: str) -> str:
        return f"{FirebasePaths.igreja(church_id)}/finance_mp_notifications"

    # --- Storage ---
    @staticmethod
    def storage_logo(church_id: str) -> str:
        return f"{FirebasePaths.igreja(church_id)}/configuracoes/logo_igreja.png"

File: flutter_app/lib/core/firebase_paths.dart
import 'package:gestao_yahweh/core/church_storage_layout.dart';
import 'package:gestao_yahweh/core/data/church_data_paths.dart';
import 'package:gestao_yahweh/core/repositories/church_repository.dart';

/// Fachada de paths Firebase (Firestore + Storage) para uso em UI/serviços.
///
/// Regra: sempre resolve [churchId] dinamicamente; nunca hardcode de tenant.
abstract final class FirebasePaths {
  FirebasePaths._();

  static String _id(String churchId) => ChurchRepository.churchId(churchId.trim());

  // --- Firestore ---
  static String igreja(String churchId) => ChurchDataPaths.churchRoot(_id(churchId));

  static String departamentos(String churchId) =>
      ChurchDataPaths.subcollection(_id(churchId), ChurchDataPaths.departamentos);

  static String departamentoDoc(String churchId, String deptId) =>
      '${departamentos(churchId)}/${deptId.trim()}';

  static String membros(String churchId) =>
      ChurchDataPaths.subcollection(_id(churchId), ChurchDataPaths.membros);

  static String configMercadoPago(String churchId) =>
      '${ChurchDataPaths.subcollection(_id(churchId), ChurchDataPaths.config)}/mercado_pago';

  static String finance(String churchId) =>
      ChurchDataPaths.subcollection(_id(churchId), ChurchDataPaths.financeiro);

  static String financeDoc(String churchId, String financeId) =>
      '${finance(churchId)}/${financeId.trim()}';

  static String financeLogs(String churchId) =>
      ChurchDataPaths.subcollection(_id(churchId), ChurchDataPaths.financeLogs);

  static String financeMpNotifications(String churchId) =>
      ChurchDataPaths.subcollection(
        _id(churchId),
        ChurchDataPaths.financeMpNotifications,
      );

  // --- Storage ---
  static String storageLogoPath(String churchId) =>
      ChurchStorageLayout.churchIdentityLogoPath(_id(churchId));
}

